### ISSUE TITLE
Can't seem to make it work...

### ISSUE THREAD

**Issue by arijbrown** [shaundon/hubot-standup-alarm#2]
Hey @shaundon - thanks for your plugin.  The theoretical functionality looks great!  I think I've got it all hooked up properly.  It responds to the list command, creation/removal of standups.  The only problem is that when the time rolls around, it never seems to actually do anything.  My usual test procedure is to create a standup 1 or 2 minutes in the future and wait to see what happens.  So far, I just get nothing.  Thoughts on what could be going wrong or how to troubleshoot?

**Comment by shaundon** [comment]
Glad you like it :)

Have you checked that you're using UTC time? It goes off the time on the server it's deployed on. 

Let me know if it's not that though and we can troubleshoot it together.

**Comment by arijbrown** [comment]
I did check with the time command - it's as you'd expect. What's troubleshooting step two?

**Comment by shaundon** [comment]
Are you using the latest version of the repo? This repo (_shaundon/hubot-standup-alarm_) was moved to the _hubot-scripts_ organisation recently, and the code on there is more recent ([here's a link](https://github.com/hubot-scripts/hubot-standup-alarm)) so I'd recommend switching to that one if you aren't already using it.

I have a feeling that it could be an issue with Hubot's `messageRoom` functionality - when you're doing things like creating standups, the replies Hubot sends you are coming from the `msg.reply` function, whereas standup messages are spontaneous messages, which are sent with `robot.messageRoom`. So it might be that it isn't working for your particular IM client. Which IM client are you using? I know that the  _hubot-scripts_ version of this repo definitely has some updates on that front, so let me know if switching to that doesn't work.

Let me know how you get on!

**Comment by arijbrown** [comment]
Yep - the repo I installed was via npm, just the other day, so I'm pretty sure I've got what there is to get.

As for clients, I'm using the Mac atlassian Hipchat client - I can try it through the web client if you think that might work.

**Comment by shaundon** [comment]
It should work with Hipchat - I was actually using Hipchat when I initially wrote it. 

Any chance you could try running it from source, instead of NPM, and dropping a couple of `console.log` statements in, for when it's checking whether to send messages, and trying to send them?

**Comment by arijbrown** [comment]
I tried but failed to get it running from non-npm install methods.  I reinstalled it via npm and watched the logs.  When I query the time (or add/remove standups), I successfully get something like this:

<pre><code>[Fri May 01 2015 14:13:52 GMT-0700 (PDT)] DEBUG  OUT > <message to="1_aris_test@conf.btf.hipchat.com/Hubot the Pony"
type="groupchat"><body>Server time is: Fri May 01 2015 14:13:52 GMT-0700 (PDT)</body></message>
</code></pre>

When hubot tries to alert the room, I get something like this:

<pre><code>
[Fri May 01 2015 14:11:01 GMT-0700 (PDT)] DEBUG  OUT > <message to="aris_test" type="chat" from="1_322@chat.btf.hipch
at.com/hubot-hipchat"><inactive xmlns="http://jabber/protocol/chatstates"/><body>Standup time. Get up, humans</body><
/message>
</code></pre>

The first shows fine, the latter ends up in the logs but I see nothing.  Is this what you'd expect?  Anything we can do anything with?

**Comment by shaundon** [comment]
Could you try reinstalling it from NPM now? I didn't realise, but the version on there was out of date.

Sorry for the delay in responding too, I've been away.

**Comment by arijbrown** [comment]
Tried reinstalling this morning and behavior appears to be the same.  Install goes smoothly, all commands seem to work (create standups, list standups, see server time, etc), but when the time comes for the alert to fire, nothing happens.  Is there an easy way for me to check the installed version number to confirm I've got the most recent bits?

**Comment by shaundon** [comment]
Sure - go to `node_modules`, then `hubot-standup-alarm`, then open `package.json`. The `version` field should be `0.0.6` (as shown on [the NPM page](https://www.npmjs.com/package/hubot-standup-alarm)).

If you're not on the right version, upgrade it with the following steps:
1. Go to `package.json` in the root folder of your Hubot, and change the `hubot-standup-alarm` version either to `*` or `>=0.0.6`.
2. Delete the `hubot-standup-alarm` folder in `node_modules`.
3. Run `npm install` to force it to download again.

Let me know how that goes!

**Comment by arijbrown** [comment]
Hrm - well I've confirmed I've got 0.0.6, so that's good and bad.  Not sure how else to debug this.  It all works except for Hubot's ability to broadcast a message.  The only thing I can tell, and I'm not sure if this is a red herring, is that all the other interactions seem to be 1:1 communication, but the standup alarm itself is broadcasting to a room and that one never happens.

**Comment by shaundon** [comment]
Sorry again for the delay - you know how it is trying to find time for side projects!

It looks like somebody else is having [the same issue](https://github.com/hubot-scripts/hubot-standup-alarm/issues/14) as you - so it looks like the `robot.messageRoom` part isn't working for Hipchat.

I'll try and take a look as soon as I can, or feel free to take a look and submit a PR!
